docs/frontend/webpack-tree-shaking.md

```diff
@@ -0,0 +1,232 @@
+transformation process for [ES6 modules](http://www.2ality.com/2014/09/es6-modules-final.html)
+whereby ESnext `export`s that are not used in a Webpack bundle can be isolated
+during code bundling and removed entirely by Uglify dead code elimination.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Basic Example](#basic-example)
+- [Tree Shaking Example](#tree-shaking-example)
+- [Advantages](#advantages)
+- [Disadvantages](#disadvantages)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+##### Basic Example
+
+(Example source available at: [github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/src/es6](https://github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/src/es6))
+
+Let's start with a very basic ESnext file that exports two functions -
+`red()` and `blue()`:
+
+[`util.js`](../../examples/frontend/src/es6/util.js)
+
+```js
+export const red = (id, msg) => `<h1 id="${id}" style="color: red">${msg}</h1>`;
+export const blue = (id, msg) => `<h1 id="${id}" style="color: blue">${msg}</h1>`;
+```
+
+Our applications then use either `blue()` or `red()` but not both:
+
+[`app1.js`](../../examples/frontend/src/es6/app1.js)
+
+```js
+import { red } from "./util";
+
+document.querySelector("#content").innerHTML += red("app1", "App 1");
+```
+
+[`app2.js`](../../examples/frontend/src/es6/app2.js)
+
+```js
+import { blue } from "./util";
+
+document.querySelector("#content").innerHTML += blue("app2", "App 2");
+```
+
+##### Tree Shaking Example
+
+(Example build / dist code available at: [github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-tree-shaking](https://github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-tree-shaking))
+
+Our goal with tree shaking is to have `app1` include the code for `red()`, but
+not `blue()` and have the opposite situation for `app2`. To accomplish this,
+we need some specific tweaks to our Webpack configuration found at:
+[`webpack.config.js`](../../examples/frontend/webpack-tree-shaking/webpack.config.js)
+
+First, our ESnext code needs the `babel-loader` to build and we need to pass
+the specific option of `modules: false` to correctly enable tree shaking:
+
+```js
+module: {
+  loaders: [
+    {
+      test: /\.js$/,
+      include: [path.join(__dirname, "../src/es6")],
+      loader: "babel",
+      query: {
+        presets: [
+          [
+            "es2015",
+            {
+              "modules": false
+            }
+          ]
+        ]
+      }
+    }
+  ]
+}
+```
+
+Next, for our compression options, we have some "real" production optimizations
+as well as a few "demo only" options for purposes of this tutorial (with a
+`DEMO ONLY` comment):
+
+```js
+plugins: [
+  new webpack.LoaderOptionsPlugin({
```

Comment:
It's new to Webpack2 -- The website should be updated with new webpack2 stuff presumably when it exits beta: https://gist.github.com/sokra/27b24881210b56bbaff7#loader-options--minimize